core/src/main/java/io/cucumber/core/plugin/PrettyFormatter.java

```diff
@@ -127,6 +131,21 @@ private void printStep(TestStepFinished event) {
                 formats.get(status + "_arg"), testStep.getDefinitionArgument());
             String locationIndent = calculateLocationIndent(event.getTestCase(), formatPlainStep(keyword, stepText));
             out.println(STEP_INDENT + formattedStepText + locationIndent + formatLocation(testStep.getCodeLocation()));
+
+            StepArgument stepArgument = testStep.getStep().getArgument();
+            if (DataTableArgument.class.isInstance(stepArgument)) {
+                DataTableFormatter tableFormatter = DataTableFormatter
+                        .builder()
+                        .prefixRow(STEP_SCENARIO_INDENT)
+                        .escapeDelimiters(false)
+                        .build();
+                DataTableArgument dataTableArgument = (DataTableArgument) stepArgument;
+                try {
+                    tableFormatter.formatTo(DataTable.create(dataTableArgument.cells()), out);
```

Comment:
the code structure is a bit different to how the rest of the step is being printed (`out.print(...)` on line 133 vs `tableFormatter.formatTo(...out)` on line 144. Considered formatting the DataTable to a string so I could follow the same pattern but it meant only the first row of the table was getting the right indent.

Let me know if it's worth restructuring the rest of `PrettyFormatter.printStep` to make this a bit more streamlined.

Reply:
No looks good.